### PR TITLE
Nix integration

### PR DESCRIPTION
**Pull request by lukasz-golebiewski**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,25 +11,24 @@
 # Choose a build environment
 dist: xenial
 
-# Do not choose a language; we provide our own build tools.
-language: generic
+language: nix
+nix: 2.3.1
+
+before_cache:
+- mkdir -p $HOME/nix.store
```
Comment (rabbitonweb) [inline .travis.yml]: It will not work without this line?

File: .travis.yml [hunk 2]
```diff
@@ -11,25 +11,24 @@
...
-    packages:
-      - libgmp-dev
+  - $HOME/nix.store
 
 before_install:
-# Download and unpack the stack executable
-- mkdir -p ~/.local/bin
-- export PATH=$HOME/.local/bin:$PATH
-- travis_retry curl -L https://get.haskellstack.org/stable/linux-x86_64.tar.gz | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'
+- sudo mkdir -p /etc/nix
```
Comment (rabbitonweb) [inline .travis.yml]: Maybe a comment what those lines are doing?

File: shell.nix [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{ghc}:
```
Comment (rabbitonweb) [inline shell.nix]: Do we need shell.nix? Why is it here. I'm asking because I think defaults are OK for now. Or do we need this so that Travis works?

**Comment by dzajkowski** [comment]
runs on osx.

**Pull request merged**